### PR TITLE
fix: `.to_public()` now doesn't underline for pyright

### PR DESCRIPTION
**Pull request by cyberglot**
`.to_public()` now doesn't underline for pyright plus some other typing improvements. It makes `.to_public()` an idempotent function if the value is already public.

### CODE REVIEW

File: nada_dsl/nada_types/scalar_types.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 """The Nada Scalar type definitions."""
 
 from dataclasses import dataclass
-from typing import Union
+from typing import Any, Self, Union, TypeVar
```
Comment (navasvarela) [inline nada_dsl/nada_types/scalar_types.py:5]: Self is a feature added in 3.11, we cannot use this feature until we move up the minimum version.

Comment (cyberglot) [reply nada_dsl/nada_types/scalar_types.py:5]: I think it should be enough to import it from `typing_extensions`, but I haven't tested it yet -- my python dev environment is extremely finicky.

Comment (navasvarela) [reply nada_dsl/nada_types/scalar_types.py]: well, you should add it to the list of dependencies in pyproject.toml then. I didn't see the change and therefore I assumed you weren't using anything new.

Comment (navasvarela) [reply nada_dsl/nada_types/scalar_types.py]: Hopefully that would fix the build

**Review by pablojhl — APPROVED** [review]

**Review by Jmgr — APPROVED** [review]

**Review by navasvarela — APPROVED** [review]
Thanks! Please make sure to increase the release version and add `typing_extensions` to `[tool.uv]`.

**Pull request merged**